.github/workflows/e2e-test.yml

```diff
@@ -0,0 +1,57 @@
+name: E2E Testing for CLI
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    types: [synchronize, opened, reopened, edited]
+    branches: [main]
+  schedule:
+    # check daily to notice potential package manager issues
+    - cron: '0 1 * * *' # At 01:00 daily
+
+jobs:
+  e2e:
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 10
+    env:
+      DEBUG: true
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macOS-latest, windows-latest]
+        node-version: ['12.x', '*']
+      fail-fast: false
+
+    steps:
+      - name: Git checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Setup node
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: 'npm'
+          cache-dependency-path: 'npm-shrinkwrap.json'
+          check-latest: true
+      - name: Install dependencies
+        run: npm ci --no-audit
+      - name: Install pnpm
+        run: npm install -g pnpm
+      - run: npm run e2e
+      - name: Notify Slack
+        uses: 8398a7/action-slack@v3
+        if: failure()
```

Comment:
We probably want to send this notification only if this is a cron job

Reply:
Maybe `if: github.event_name == 'schedule' will work

Reply:
I thought we want to have all failings? 🤔 maybe only if branch is main?

Reply:
because schedule is running on main as well

Reply:
For PRs and `main` we should see those as failed status checks and can also make them required checks.

For Slack notifications, we'd want to capture the ones that are caused by a breaking change to one of our dependencies like we had with `colors.js`.

My concern is that if every failure is reported to the Slack channel it might get too busy.
 
 We can go with send a notification if it fails on `main` and adjust accordingly, WDYT?

Reply:
Going with your proposed schedule check :) if we oversee something we can always get more restrictive but I agree it should not get noisy